sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -276,6 +278,29 @@ def transform(self, X):
         return sp.csr_matrix(X)
 
 
+class BinaryDecisionTreeClassifier(DecisionTreeClassifier):
+    def _more_tags(self):
+        return {'binary_only': True}
+
+    def fit(self, X, y, sample_weight=None):
+        super().fit(X, y, sample_weight)
+        X, y = check_X_y(X, y, y_numeric=False)
+        check_classification_targets(y)
+        self.classes_, y = np.unique(y, return_inverse=True)
```

Comment:
can you directly use `self.n_classes_` from `DecisionTreeClassifier` and make `fit` as short as possible?

Also please add a small docstring to the class, e.g. "Toy classifier that only support binary classification"

Reply:
Yep, done